src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -104,6 +133,52 @@ def handle_result_for_project(
             ex=LAST_UPDATE_REDIS_TTL,
         )
 
+    def handle_result_for_project_auto_onboarding_mode(
+        self, project_subscription: ProjectUptimeSubscription, result: CheckResult
+    ):
+        if result["status"] == CHECKSTATUS_FAILURE:
+            redis = _get_cluster()
+            key = build_onboarding_failure_key(project_subscription)
+            pipeline = redis.pipeline()
+            pipeline.incr(key)
+            pipeline.expire(key, ONBOARDING_FAILURE_REDIS_TTL)
+            failure_count = pipeline.execute()[0]
+            if failure_count >= ONBOARDING_FAILURE_THRESHOLD:
+                # If we've hit too many failures during the onboarding period we stop monitoring
+                delete_project_uptime_subscription(
+                    project_subscription.project,
+                    project_subscription.uptime_subscription,
+                    modes=[ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING],
+                )
+                # Mark the url as failed so that we don't attempt to auto-detect it for a while
+                set_failed_url(project_subscription.uptime_subscription.url)
+                redis.delete(key)
+                metrics.incr("uptime.result_processor.autodetection.failed_onboarding")
+        elif result["status"] == CHECKSTATUS_SUCCESS:
+            assert project_subscription.date_added is not None
+            if (
+                datetime.now(timezone.utc) - ONBOARDING_MONITOR_PERIOD
```

Comment:
you can also use `timezone.now()` from django, idk what we prefer

Reply:
Yeah that's probably preferable, I can't keep all the different ways of creating dates straight these days

Reply:
Actually, this should be the time the result was generated, not the wall clock time. Doesn't matter a lot probably, but it's more correct

Reply:
Makes sense